src/wallet/walletdb.cpp

```diff
@@ -792,6 +782,25 @@ static DBErrors LoadLegacyWalletRecords(CWallet* pwallet, DatabaseBatch& batch,
     });
     result = std::max(result, pool_res.m_result);
 
+    // Deal with old "wkey" and "defaultkey" records.
+    // These are not actually loaded, but we need to check for them
+
+    // We don't want or need the default key, but if there is one set,
+    // we want to make sure that it is valid so that we can detect corruption
+    CPubKey default_pubkey;
+    if (batch.Read(DBKeys::DEFAULTKEY, default_pubkey) && !default_pubkey.IsValid()) {
+        pwallet->WalletLogPrintf("Error reading wallet database: Default Key corrupt\n");
+        return DBErrors::CORRUPT;
+    }
+
+    // "wkey" records are unsupported, if we see any, throw an error
+    LoadResult wkey_res = LoadRecords(pwallet, batch, DBKeys::OLD_KEY,
+        [] (CWallet* pwallet, DataStream& key, CDataStream& value, std::string& err) {
+        pwallet->WalletLogPrintf("Found unsupported 'wkey' record, try loading with version 0.18\n");
+        return DBErrors::LOAD_FAIL;
```

Comment:
nit: could remove the logging and let `LoadRecords` log it:
```c++
err = "Found unsupported 'wkey' record, try loading with version 0.18";
return DBErrors::LOAD_FAIL;
```

Reply:
Done